### PR TITLE
Applied cors instructions for setting up fauxton locally for dev

### LINKED ISSUE #1202: Content Security Policy violation errors following local development setup when attempting to start dev
I have been following the guide(s) - `readme.md` and read the guide on contributing. When I execute `npm run dev` with couchdb up 'n running in docker and the `settings.json` modified accordingly, I get a Content Security violation in `Chrome Version 75.0.3770.100 (Official Build) (64-bit)` on the latest `Mac Book Pro` as of this writing:

```
Refused to connect to '<URL>' because it violates the following Content Security Policy directive: "default-src 'self'". Note that 'connect-src' was not explicitly set, so 'default-src' is used as a fallback.
```

`default-src 'self'` should work for same-origin when applying host: `"http://localhost:5984"`

I was playing around with https://csp-evaluator.withgoogle.com/ using the `defaultHeaderValue` in `devserver.js` but to no avail. (On an unrelated note, Google is saying you can restrict `object src to 'none'` .. new issue?

This is blocking me from completing local development setup and it would be my pleasure to contribute and fix some UI issues I found.

### PR DESCRIPTION
**Pull request by iamgollum**
Resolves #1202

### CODE REVIEW

**Comment by popojargo** [comment]
@iamgollum I  added few comments . Let me know what you think

**Comment by iamgollum** [comment]
@popojargo all for it. Updating now

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -57,26 +56,40 @@ Now run:
 
 And your Fauxton dev server will be up and running at `localhost:8000`.
 
+Caveats:
```
Comment (garrensmith) [inline CONTRIBUTING.md]: Weird. I haven't worked on Fauxton in awhile but this shouldn't be required. Fauxton dev mode runs through a local proxy. It just expects a CouchDB instance at `localhost:8000`

Comment (popojargo) [reply CONTRIBUTING.md]: In the past, I had the same problem when working with attachments. I don't know if its still the case

Comment (iamgollum) [reply CONTRIBUTING.md]: @garrensmith @popojargo When I start dev this time around it clears out the CORS configuration set by the plugin and the CORS config in the UI is inaccessible:

Screen Shot 2019-07-10 at 7 28 55 PM

**Comment by popojargo** [comment]
Kinf of usual but I had the same warnings with attachments

On Jul. 10, 2019 7:30 p.m., Aaron Tobias <notifications@github.com> wrote:

@iamgollum commented on this pull request.

________________________________

In CONTRIBUTING.md<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fapache%2Fcouchdb-fauxton%2Fpull%2F1204%23discussion_r302312151&data=02%7C01%7C%7Cc34ee0e897df4ec9796c08d7058e874d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636983982017743642&sdata=KreBrjJRQMkkWQ6uiGnBXtDty0CVANypVryoROLqgCA%3D&reserved=0>:

> @@ -57,26 +56,40 @@ Now run:

 And your Fauxton dev server will be up and running at `localhost:8000`.

+Caveats:


@garrensmith<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgarrensmith&data=02%7C01%7C%7Cc34ee0e897df4ec9796c08d7058e874d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636983982017753653&sdata=zXaTHRQfkExa%2BsgAn0B2k32TGohp%2BLBn5GYsytryN5w%3D&reserved=0> @popojargo<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpopojargo&data=02%7C01%7C%7Cc34ee0e897df4ec9796c08d7058e874d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636983982017753653&sdata=SXjDPDW4plaUx%2FFtqRc6U%2BHcwe5HtF1A19HVZOpDRsg%3D&reserved=0> When I start dev this time around it clears out the CORS configuration set by the plugin and the CORS config in the UI is inaccessible:

[Screen Shot 2019-07-10 at 7 28 55 PM]<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F1408766%2F61011571-17dfe500-a349-11e9-93a8-a54debca2e36.png&data=02%7C01%7C%7Cc34ee0e897df4ec9796c08d7058e874d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636983982017763670&sdata=KNJt9idt%2F9ajNnRDzj0BuL5%2Bxhr5V%2FPAxD10iN4qYtE%3D&reserved=0>

**Pull request merged**